tests/test_reserveSourcesTask.py

```diff
@@ -0,0 +1,154 @@
+        key = schema[name + "_reserved"].asKey()
+        catalog = lsst.afw.table.SourceCatalog(schema)
+        catalog.reserve(self.num)
+        for _ in range(self.num):
+            catalog.addNew()
+        return Struct(catalog=catalog, task=task, key=key)
+
+    def check(self, sources, task, key, fraction):
+        """Check that source reservation is working
+
+        We test that the source reservation works, that it depends
+        on the RNG seed and that thigns behave as we expect when there's
+        a prior selection.
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.Catalog` or `list` of `lsst.afw.table.Record`
+            List of sources.
+        task : `lsst.meas.algorithms.ReserveSourcesTask`
+            Task to do the reservations.
+        key : `lsst.afw.table.Key`
+            Key to the flag column.
+        fraction : `float`
+            Fraction of sources to reserve.
+        """
+        numExpect = fraction*len(sources)
+
+        # No prior
+        results1 = task.run(sources, expId=1)
+        flagged1 = np.array([ss.get(key) for ss in sources])
+        self.assertEqual(flagged1.sum(), numExpect)
+        np.testing.assert_array_equal(results1.reserved, flagged1)
+        np.testing.assert_array_equal(results1.use, np.logical_not(flagged1))
+
+        # Second run with different seed
+        for ss in sources:
+            ss.set(key, False)
+        results2 = task.run(sources, expId=2)
+        flagged2 = np.array([ss.get(key) for ss in sources])
+        self.assertEqual(flagged1.sum(), numExpect)
+        np.testing.assert_array_equal(results2.reserved, flagged2)
+        np.testing.assert_array_equal(results2.use, np.logical_not(flagged2))
+        if numExpect > 0:
+            self.assertFalse(np.all(flagged1 == flagged2))  # Pretty unlikely they match given different seeds
+
+        # Run with prior selection
+        for ss in sources:
+            ss.set(key, False)
+        prior = np.arange(0, self.num, 1, dtype=int) % 2 == 0
+        results3 = task.run(sources, prior, expId=1)
+        flagged3 = np.array([ss.get(key) for ss in sources])
+        self.assertEqual(flagged3.sum(), fraction*prior.sum())
+        if numExpect > 0:
+            self.assertFalse(np.all(flagged1 == flagged3))  # Flagging should change, despite same seed
+        np.testing.assert_array_equal(results3.reserved, flagged3)
+        self.assertEqual((results3.use & flagged3).sum(), 0)  # No sources should be both reserved and used
+        self.assertEqual(results3.use.sum(), (1.0 - fraction)*prior.sum())
+        self.assertEqual(results3.reserved.sum(), fraction*prior.sum())
+        np.testing.assert_array_equal(results3.use, prior & ~flagged3)  # Actual definition of 'use'
+
+    def testCatalog(self):
+        """Test source reservation with a Catalog
+
+        We test multiple reservation fractions.
+        """
+        for fraction in (0.0, 0.1, 0.2, 0.5):
```

Comment:
Why not `for fraction in np.linspace(0,1,25)` to catch a variety of different fractions?

Reply:
The choice of fractions is so that the multiplications produce whole numbers, and I thought doing 3 non-zero fractions was sufficient.

Reply:
But I worry that you aren't testing the non-int multiplication case, which the code should be able to handle, no?